include/graphics/graphics.hpp

```diff
@@ -20,60 +21,88 @@ class Graphics {
     class GameEvent {
       public:
         enum class EventType {
+            no_event,
             place_card_to_base,
             place_card_to_deck,
             show_deck,
             show_base,
             end_turn,
             close_window,
             wait_event,
-            no_event,
         };
 
       public:
-        EventType type;
-        const PlayerCard *card;
-        const Base *base;
-        const Deck<PlayerCard *> *deck;
+        EventType type = EventType::no_event;
+        const PlayerCard *card = nullptr;
+        const Base *displayed_base = nullptr;
+        const Deck<PlayerCard *> *displayed_deck = nullptr;
+        const Base *base = nullptr;
+        const Deck<PlayerCard *> *deck = nullptr;
 
       public:
-        GameEvent(EventType type_ = EventType::no_event, const PlayerCard *card_ptr = nullptr,
-                  const Base *base_ptr = nullptr, const Deck<PlayerCard *> *deck_ptr = nullptr)
-            : type(type_), card(card_ptr), base(base_ptr), deck(deck_ptr) {}
+        explicit GameEvent() = default;
+        explicit GameEvent(EventType Type, const PlayerCard *Card = nullptr,
+                           const Base *Displayed_base = nullptr, const Deck<PlayerCard *> *Displaed_deck = nullptr,
+                           const Base *Base = nullptr, const Deck<PlayerCard *> *Deck = nullptr)
+            : type(Type), card(Card), displayed_base(Displayed_base), displayed_deck(Displaed_deck), base(Base),
+              deck(Deck) {}
+        explicit GameEvent(const GameEvent &) = delete;
+        explicit GameEvent(GameEvent &&) = delete;
+        const GameEvent& operator=(const GameEvent &rhs);
+        const GameEvent& operator=(GameEvent &&rhs);
         ~GameEvent() = default;
     };
 
-  public:
-    constexpr static const float player_scale_to_playground = 0.38;
-    constexpr static const float player_increase = 1.8;
-    constexpr static const float base_shift_to_playground = 0.1;
-    constexpr static const float card_scale_to_player = 0.2;
-    constexpr static const float bases_place_to_playground = 0.8;
-    constexpr static const float cards_place_to_player = 0.6;
-    constexpr static const float bases_pos_to_playground = 0.25;
+    class LocationSettings {
+      public:
+        constexpr static const float player_scale_to_playground_x = 0.38;
+        constexpr static const float player_scale_to_playground_y = 0.25;
```

Comment:
Same for other line in this function

Reply:
It is necessary because it constantly sets the location of the player, and therefore the cards.